### PR TITLE
collins_client clean up

### PR DESCRIPTION
**Pull request by defect**
Basically doing the same thing i did for collins_state. Get rid of jeweler, cleanup the gemspec, bump a couple of gem versions. I've intentionally kept httparty at 1.11.0 since in later versions they drop support for ruby 1.9.2. This should work for the case in https://github.com/tumblr/collins/pull/204 since the gem conflicting there only requires httparty `~> 1.11.0`.

I'm also updating a bunch of tests to fix deprecations warnings. The tests run cleanly on 1.9.2, 1.9.3, and 2.2.

@byxorna @Primer42 @maddalab @roymarantz

### CODE REVIEW

File: support/ruby/collins-client/Gemfile.lock [hunk 1]
```diff
@@ -1,50 +0,0 @@
-GEM
```
Comment (byxorna) [inline support/ruby/collins-client/Gemfile.lock]: We should keep the gemfile.lock in versioncontrol, and just run a `bundle update`. That way we know the exact versions of things that work when we are devving.

Comment (defect) [inline support/ruby/collins-client/Gemfile.lock]: If we want exact versions of gems we depend on I think those should go in the gemspec. I can add some pessimistic version requirements to the gemspec based on what I've been using when testing it on the various versions of ruby.

**Comment by roymarantz** [comment]
:+1:  if this really works :-)

File: support/ruby/collins-client/Gemfile.lock (hunk 1 above)
Comment (Primer42) [inline support/ruby/collins-client/Gemfile.lock]: I agree with @defect - for gems you should not have Gemfile.lock checked in.

File: support/ruby/collins-client/collins_client.gemspec [hunk 2]
```diff
@@ -1,8 +1,3 @@
-# Generated by jeweler
-# DO NOT EDIT THIS FILE DIRECTLY
-# Instead, edit Jeweler::Tasks in Rakefile, and run 'rake gemspec'
-# -*- encoding: utf-8 -*-
-
 Gem::Specification.new do |s|
   s.name = "collins_client"
   s.version = File.read 'VERSION'
```
Comment (Primer42) [inline support/ruby/collins-client/collins_client.gemspec]: I think you need to do a version bump still

**Comment by Primer42** [comment]
LGTM :ship: :it:



(the ship emojii, and then the it emoji...... get it?)

**Pull request merged**